memiavl/multitree.go

```diff
@@ -65,20 +65,9 @@ func NewEmptyMultiTree(initialVersion uint32, cacheSize int) *MultiTree {
 }
 
 func LoadMultiTree(dir string, zeroCopy bool, cacheSize int) (*MultiTree, error) {
-	// load commit info
-	bz, err := os.ReadFile(filepath.Join(dir, MetadataFileName))
+	metadata, err := readMetadata(dir)
 	if err != nil {
-		return nil, err
-	}
-	var metadata MultiTreeMetadata
-	if err := metadata.Unmarshal(bz); err != nil {
-		return nil, err
-	}
-	if metadata.CommitInfo.Version > math.MaxUint32 {
-		return nil, fmt.Errorf("commit info version overflows uint32: %d", metadata.CommitInfo.Version)
-	}
-	if metadata.InitialVersion > math.MaxUint32 {
-		return nil, fmt.Errorf("initial version overflows uint32: %d", metadata.InitialVersion)
+		return nil, fmt.Errorf("fail to read metadata: %w", err)
```

Comment:
seems we can't wrap error here since we check with [os.IsNotExist](https://github.com/crypto-org-chain/cronos/blob/d0c70fd09c28181df7abd18bfc362fddd0ec30f7/memiavl/db.go#L122)
```suggestion
		return nil, err
```